src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/EngineGeneralTest.java

```diff
@@ -4121,34 +4122,35 @@ public void entityEntryHashcodeTest() {
      * exited from.
      *
      * @since 2.1.12
+     * @throws Exception if something goes wrong while checking if the connection of the engine is closed.
      */
     @Test
-    public void tryWithResourcesClosesEngine() {
```

Comment:
This test was relying on `MockUp.getMockInstance()`, which doesn't exist anymore on the newest version.

Initially I removed it, since it was just verifying Java functionality and not PDB, and was equivalent to simply doing
`assertTrue(AutoCloseable.class.isAssignableFrom(DatabaseEngine.class))`

I restored the test because we can test that the DatabaseEngine behaves as expected in a try-with-resources,
by checking that it closes the underlying DB connection.